mlir/lib/Dialect/MemRef/Utils/MemRefUtils.cpp

```diff
@@ -75,18 +74,31 @@ std::pair<LinearizedMemRefInfo, OpFoldResult> getLinearizedMemRefOffsetAndSize(
     addMulMap = addMulMap + symbols[offsetIdx] * symbols[offsetIdx + 1];
     offsetValues[offsetIdx] = indicesVec[i];
     offsetValues[offsetIdx + 1] = strides[i];
-
-    mulMap = mulMap * symbols[i];
   }
-
   // Adjust linearizedIndices and size by the scale factor (dstBits / srcBits).
   int64_t scaler = dstBits / srcBits;
-  mulMap = mulMap.floorDiv(scaler);
-
   OpFoldResult linearizedIndices = affine::makeComposedFoldedAffineApply(
       builder, loc, addMulMap.floorDiv(scaler), offsetValues);
+
+  size_t symbolIndex = 0;
+  SmallVector<OpFoldResult> values;
+  SmallVector<AffineExpr> productExpressions;
+  for (unsigned i = 0; i < sourceRank; ++i) {
+    AffineExpr strideExpr = symbols[symbolIndex++];
+    OpFoldResult stride = strides[i];
+    values.push_back(getValueOrCreateConstantIndexOp(builder, loc, stride));
```

Comment:
```suggestion
    values.push_back(strides[i]);
```

Reply:
Reasonable simplification, it doesn't need to be `Value` type given that everything else is `OpFoldResult`.